### ISSUE TITLE
GPyOpt BayesianOptimization Fails to complete execution after initial_design_numdata iterations.

### ISSUE THREAD

**Issue by amohar2** [SheffieldML/GPyOpt#99]
I am trying to use the parallel GPyOpt using the instructions on the following notebook:
http://nbviewer.jupyter.org/github/SheffieldML/GPyOpt/blob/master/manual/GPyOpt_parallel_optimization.ipynb

The notebook itself runs without any issues. In my case, I only change the objective function to a Neural Network using Keras library (with TensorFlow backend). And consequently, I change the hyper-parameters' domain into something like this:
    domain =[{'name': 'num_hidden_layers', 'type': 'discrete', 'domain': (1,2,3)},
                   {'name': 'num_neurons', 'type': 'discrete', 
                    'domain': tuple(range(1,21)),'dimensionality': 3},
               {'name': 'learning_rate', 'type': 'discrete', 
                'domain': tuple(np.logspace(np.log10(0.001), np.log10(1), num=10))},
               {'name': 'l2_reg', 'type': 'discrete', 
                'domain': tuple(np.logspace(np.log10(0.001), np.log10(1), num=10))},
               {'name': 'batch_size', 'type': 'discrete', 'domain': tuple(range(1,51))},
                  {'name': 'dropout', 'type': 'continuous', 
                   'domain': (0.01,0.99), 'dimensionality': 3}]

This has mixed_type variables.

But for some reason, After I call the BayesianOptimization method as follows:
num_cores, batch_size = 1, 1
GPyOpt.methods.BayesianOptimization(f=fcnet_regression_loss,  (this outputs neural network validation accuracy)
                                            domain = domain,                  
                                            acquisition_type = 'EI',              
                                            normalize_Y = True,
                                            initial_design_numdata = 10,
                                          evaluator_type = 'local_penalization',
                                            batch_size = batch_size,
                                            num_cores = num_cores,
                                            acquisition_jitter = 0) 

After 10 iterations, the notebook crashes and shuts down the Jupyter kernel.

I couldn't still figure out what causes this problem.

**Comment by amohar2** [comment]
If I change the domain into 'continuous' type variables only, i.e.,
    mixed_domain =[{'name': 'num_hidden_layers', 'type': 'continuous', 'domain': (0.9,3.1)},
                   {'name': 'num_neurons', 'type': 'continuous', 
                    'domain': (1,21),'dimensionality': 3},
               {'name': 'learning_rate', 'type': 'continuous', 
                'domain': (0.1,0.9)},
               {'name': 'l2_reg', 'type': 'continuous', 
                'domain': (0.1,0.9)},
               {'name': 'batch_size', 'type': 'continuous', 'domain': (5,51)},
                  {'name': 'dropout', 'type': 'continuous', 
                   'domain': (0.01,0.99), 'dimensionality': 3}]

Then the optimization converges.
Also, for mixed type domains, with both discrete and continuous type variables, if I choose the discrete sets very small (in order of two or three elements each) then algorithm converges, but it is very slow.

**Comment by javiergonzalezh** [comment]
In the new version the issue with being categorical variables very slow has been solved. Please let us know if you have problems with it.

**Comment by timbryant** [comment]
Hi,there @amohar2 
Thank you for your work,I'm interested in your work that you changed the objective function to a Neural Network using Keras library,could you tell me how did you define the fcnet_regression_loss? Because I wrote a func,which is used for optimize the Neural Network,but it did not work.
Looking forward to your apply
thank you very much
